mtc.go

```diff
@@ -369,68 +321,55 @@ func (c *BikeshedCertificate) MarshalBinary() ([]byte, error) {
 	var b cryptobyte.Builder
 	buf, err := c.Assertion.MarshalBinary()
 	if err != nil {
-		return nil, fmt.Errorf("Failed to marshal Assertion: %w", err)
+		return nil, fmt.Errorf("failed to marshal Assertion: %w", err)
 	}
 	b.AddBytes(buf)
-	b.AddUint16(uint16(c.Proof.TrustAnchor().ProofType()))
-	b.AddUint8LengthPrefixed(func(b *cryptobyte.Builder) {
-		b.AddBytes(c.Proof.TrustAnchor().Info())
-	})
+
+	buf, err = c.Proof.TrustAnchorIdentifier().MarshalBinary()
+	b.AddBytes(buf)
```

Comment:
Length prefix is missing

Reply:
The `MarshalBinary()` method of the `tai` does encode the OID with the TLS style length prefix. From how I understand the I-D, there is no other, outer length encoding needed. See [Section 5.5.3](https://datatracker.ietf.org/doc/html/draft-davidben-tls-merkle-tree-certs-03#section-5.5.3) and [`mct.go:1603`](https://github.com/pohlm01/mtc/blob/main/mtc.go#L1603) for reference.

Reply:
The definition is

```
opaque TrustAnchorIdentifier<1..2^8-1>
```

As you can read from [section 4.3 from RFC 5246](https://datatracker.ietf.org/doc/html/rfc5246#section-4.3), the angular brackets indicate a length prefixed value.

Reply:
Agreed. Maybe I'm just confusing something here, but in my mind, this is taken care of in [`mct.go:1603`](https://github.com/pohlm01/mtc/blob/main/mtc.go#L1603). Maybe it becomes more clear if I move this length prefix out of the TAI `MarshalBinary()` method?

Reply:
Oh, you're right. Now my complain becomes the opposite: there are a few places where a second length prefix is included.